src/elements/content-sidebar/versions/VersionsSidebarContainer.js

```diff
@@ -82,22 +118,118 @@ class VersionsSidebarContainer extends React.Component<Props, State> {
+    handleFetchSuccess = ([file, versions]): [BoxItem, FileVersions] => {
         const { api } = this.props;
-        const { entries: versions } = api.getVersionsAPI(false).addCurrentVersion(responseData, file) || {};
+        const { entries } = api.getVersionsAPI(false).addCurrentVersion(versions, file) || {};
 
         this.setState({
             error: undefined,
             isLoading: false,
             permissions: file.permissions || {},
-            versions: versions.sort((a, b) => Date.parse(b.created_at) - Date.parse(a.created_at)),
+            versions: entries.sort((a, b) => Date.parse(b.created_at) - Date.parse(a.created_at)),
         });
+
+        return [file, versions];
+    };
+
+    fetchData = (): Promise<any> => {
+        return Promise.all([this.fetchFile(), this.fetchVersions()])
+            .then(this.handleFetchSuccess)
+            .catch(this.handleFetchError);
+    };
+
+    fetchFile = (options = {}): Promise<BoxItem> => {
+        const { api, fileId } = this.props;
+
+        return new Promise((resolve, reject) =>
+            api.getFileAPI().getFile(fileId, resolve, reject, {
+                fields: FILE_VERSION_FIELDS_TO_FETCH,
+                forceFetch: true,
+                ...options,
+            }),
+        );
+    };
+
+    fetchVersions = (): Promise<FileVersions> => {
+        const { api, fileId } = this.props;
+
+        return new Promise((resolve, reject) => api.getVersionsAPI(false).getVersions(fileId, resolve, reject));
+    };
+
+    deleteVersion = (versionId: string): Promise<BoxItemVersion> => {
+        const { api, fileId } = this.props;
+
+        return new Promise((resolve, reject) =>
+            api.getVersionsAPI(false).deleteVersion(fileId, versionId, resolve, reject),
+        );
+    };
+
+    promoteVersion = (versionId: string): Promise<BoxItemVersion> => {
+        const { api, fileId } = this.props;
+
+        return new Promise((resolve, reject) =>
+            api.getVersionsAPI(false).promoteVersion(fileId, versionId, resolve, reject),
+        );
+    };
+
+    fetchData = (): Promise<any> => {
+        return Promise.all([this.fetchFile(), this.fetchVersions()])
+            .then(this.handleFetchSuccess)
+            .catch(this.handleFetchError);
+    };
+
+    fetchFile = (options = {}): Promise<BoxItem> => {
+        const { api, fileId } = this.props;
+
+        return new Promise((resolve, reject) =>
+            api.getFileAPI().getFile(fileId, resolve, reject, {
+                fields: FILE_VERSION_FIELDS_TO_FETCH,
+                forceFetch: true,
+                ...options,
+            }),
+        );
+    };
+
+    fetchVersions = (): Promise<FileVersions> => {
+        const { api, fileId } = this.props;
+
+        return new Promise((resolve, reject) => api.getVersionsAPI(false).getVersions(fileId, resolve, reject));
+    };
+
+    deleteVersion = (versionId: string): Promise<BoxItemVersion> => {
```

Comment:
some of these are duplicated

Reply:
Good catch. I think I had a bad rebase when I was preparing the PR. Will fix!